storage/innobase/include/row0mysql.h

```diff
@@ -632,6 +632,7 @@ struct row_prebuilt_t {
 					(VARCHAR can be off-page too) */
 	unsigned	versioned_write:1;/*!< whether this is
 					a versioned write */
+	ulint		duplicates;	/*!< TRX_DUP_IGNORE | TRX_DUP_REPLACE */
```

Comment:
I would consider using two separate bitfields:
```c
unsigned ignore:1;
unsigned replace:1;
```
Also, remove `TRX_DUP_IGNORE` and `TRX_DUP_REPLACE` (using 0 or 1 instead) because the `TRX_` prefix is misleadingly referring to transactions.